server/src/main/java/org/apache/kafka/network/metrics/RequestMetrics.java

```diff
@@ -214,5 +215,20 @@ private DeprecatedRequestRateKey(short version, ClientInformation clientInformat
             this.version = version;
             this.clientInformation = clientInformation;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            DeprecatedRequestRateKey that = (DeprecatedRequestRateKey) o;
+            return version == that.version && Objects.equals(clientInformation, that.clientInformation);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = version;
```

Comment:
How about using `Objects.hash` to simplify the code?

Reply:
Sure, done